examples/ul2/src/super_glue/data.py

```diff
@@ -0,0 +1,262 @@
+from composer.core.types import Dataset
+from composer.utils import dist
+from omegaconf.listconfig import ListConfig
+from torch.utils.data import DataLoader
+
+log = logging.getLogger(__name__)
+
+_task_prefix_column_names = {
+    'boolq': ['passage', 'question'],
+    'cb': ['hypothesis', 'premise'],
+    'copa': ['choice1', 'choice2', 'premise', 'question'],
+    'multirc': ['question', 'answer', 'paragraph'],
+    'record': ['passage', 'query'],
+    'rte': ['hypothesis', 'premise'],
+    'wic': ['sentence1', 'sentence2', 'word'],
+    'wsc': ['text', 'span1_text', 'span2_text'],
+}
+_task_target_maps = {
+    'boolq': {
+        0: 'False',
+        1: 'True'
+    },
+    'cb': {
+        0: 'entailment',
+        1: 'not_entailment',
+        2: 'neutral'
+    },
+    'copa': {
+        0: 'choice1',
+        1: 'choice2'
+    },
+    'multirc': {
+        0: 'False',
+        1: 'True'
+    },
+    'record':
+        lambda example: ''
+        if len(example['answers']) == 0 else example['answers'][0],
+    'rte': {
+        0: 'entailment',
+        1: 'not_entailment',
+        2: 'neutral'
+    },
+    'wic': {
+        0: 'False',
+        1: 'True'
+    },
+    'wsc': {
+        0: 'False',
+        1: 'True'
+    },
+}
+
+
+def create_super_glue_dataset(
+    task: str,
+    tokenizer_name: str,
+    split: str,
+    max_seq_length: int = 256,
+    decoder_only_format: bool = False,
+    max_retries: int = 10,
+    num_workers: int = 0,
+    extra_prefix: Optional[str] = None,
+):
+    if task not in _task_prefix_column_names:
+        raise ValueError(
+            f'task ({task}) must be one of {_task_prefix_column_names.keys()}')
+
+    if (max_seq_length % 8) != 0:
+        log.warning(
+            'For performance, a max_seq_length as a multiple of 8 is recommended.'
+        )
+
+    tokenizer = transformers.AutoTokenizer.from_pretrained(
+        tokenizer_name,
+        model_max_length=max_seq_length)  #type: ignore (thirdparty)
+
+    log.info(f'Loading {task.upper()} on rank {dist.get_global_rank()}')
+    download_config = datasets.DownloadConfig(max_retries=max_retries)
+    dataset = datasets.load_dataset(
+        'super_glue',
+        task,
+        split=split,
+        download_config=download_config,
+    )
+
+    log.info(
+        f'Starting tokenization by preprocessing over {num_workers} threads!')
+    prefix_column_order = _task_prefix_column_names[task]
+    target_extractor_map = _task_target_maps[task]
+    if isinstance(target_extractor_map, dict):
+        target_extractor = lambda example: target_extractor_map[example['label']
+                                                               ]
+    else:
+        assert callable(target_extractor_map)
+        target_extractor = target_extractor_map
+
+    def convert_padding_token(input_ids: List,
+                              attention_mask: List,
+                              new_pad_id: int = -100):
```

Comment:
Looks like we need our constant to be public

```suggestion
                              new_pad_id: int = HF_IGNORE_TOKEN):
```